internal/rbd/replicationcontrollerserver.go

```diff
@@ -557,7 +557,13 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 
 	// promote secondary to primary
 	if !mirroringInfo.Primary {
-		err = rbdVol.promoteImage(req.Force)
+		if req.Force {
+			// workaround for https://github.com/ceph/ceph-csi/issues/2736
+			// TODO: remove this workaround when the issue is fixed
+			err = rbdVol.forcePromoteImage(cr)
+		} else {
+			err = rbdVol.promoteImage(req.Force)
```

Comment:
```suggestion
			err = rbdVol.promoteImage(false)
```

Reply:
once I remove the workaround I need to change this from false to req.Force. keeping it as for now. Hope its fine